src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -86,244 +128,146 @@ class Ignored(Exception):
 class Plugin(object):
     """Implements an ASAP::O producer plugin
     """
+
     def __init__(self, plugin_config):
         super().__init__()
 
         self.config = plugin_config
-        self.required_parameter = []
-
-        self.stream_info = {}
-
-        self.endpoint = None
-        self.beamtime = None
-        self.beamtime_file_path = None
-        self.beamtime_file_regex = None
-        self.token = None
-        self.token_file = None
-        self.stream = None
-        self.n_threads = None
-        self.ingest_mode = None
-        self.file_regex = None
-        self.ignore_regex = None
-        self.file_start_index = None
-
-        self.timeout = 1000
-
-        self.data_type = None
+
+        self.timeout = 1
+        self.config_timeout = 10
+        self.config_time = 0
+        self.check_time = 0
+
+        self.asapo_worker = None
         self.lock = None
         self.log = None
```

Comment:
I think it would be easier to use a global logger. There will be only one instance of the Plugin anyway.